java/src/main/c++/vw_learner_VWFactory.cc

```diff
@@ -0,0 +1,39 @@
+#include "../../../../vowpalwabbit/vw.h"
+#include "jni_base_learner.h"
+#include "vw_learner_VWFactory.h"
+
+JNIEXPORT jlong JNICALL Java_vw_learner_VWFactory_initialize(JNIEnv *env, jobject obj, jstring command)
+{ jlong vwPtr = 0;
+  try
+  { vw* vwInstance = VW::initialize(env->GetStringUTFChars(command, NULL));
+    vwPtr = (jlong)vwInstance;
+  }
+  catch(...)
+  { rethrow_cpp_exception_as_java_exception(env);
+  }
+  return vwPtr;
+}
+
+JNIEXPORT jobject JNICALL Java_vw_learner_VWFactory_getReturnType(JNIEnv *env, jobject obj, jlong vwPtr)
+{ jclass clVWReturnType = env->FindClass("vw/learner/VWReturnType");
+    jfieldID field = env->GetStaticFieldID(clVWReturnType , "Unknown", "Lvw/learner/VWReturnType;");
+    try
+    { vw* vwInstance = (vw*)vwPtr;
+        if (vwInstance->p->lp.parse_label == simple_label.parse_label) {
```

Comment:
@JohnLangford please check these if statements.  I'm not sure if they are correct and the one on line 30 is really bothering me.